src/codeu/chat/DerbyStore.java

```diff
@@ -95,6 +102,43 @@ public DerbyStore() {
 		}
 	}
 	
+	@Override
+	public User getUser(Uuid userid) throws SQLException, IOException {
+		PreparedStatement getUser = conn.prepareStatement(getUserInfo);
+		getUser.setString(1, userid.toString());
+		ResultSet user = getUser.executeQuery();
+		
+		return new User(Uuid.parse(user.getString(1)), user.getString(2), Time.fromMs(user.getLong(2)));
+	}
+	
+	@Override
+	public Conversation getConversation(Uuid conversationid) throws SQLException, IOException {
+		PreparedStatement getConversation = conn.prepareStatement(getConversationInfo);
+		getConversation.setString(1, conversationid.toString());
+		ResultSet conversation = getConversation.executeQuery();
+		
+		PreparedStatement getUsersForConversation = conn.prepareStatement(getUsersInvolved);
+		getUsersForConversation.setString(1, conversationid.toString());
+		ResultSet users = getUsersForConversation.executeQuery();
+		
+		HashSet<Uuid> usersInvolved = new HashSet<Uuid>();
+		
+		while (users.next()) {
+			usersInvolved.add(Uuid.parse(users.getString(1)));
+		}
+		
+		Uuid id = Uuid.parse(conversation.getString(1));
+		Uuid owner = Uuid.parse(conversation.getString(2));
+		Time creation = Time.fromMs(conversation.getLong(3));
+		String title = conversation.getString(4);
+		Uuid firstMessage = Uuid.parse(conversation.getString(5));
+		Uuid lastMessage = Uuid.parse(conversation.getString(6));
+				
+		return new Conversation(id, owner, creation, title, usersInvolved, firstMessage, lastMessage);
```

Comment:
Don't feel like you have to stick with the design you were given: for example, do we still need firstMessage and lastMessage? Do we still need usersInvolved? (I don't know the answer to those questions, but the point is that you can change anything you want, depending on exactly what the new client will need.)